homeassistant/components/mobile_app/webhook.py

```diff
@@ -48,14 +48,26 @@ def register_deleted_webhooks(hass: HomeAssistantType, store: Store):
             pass
 
 
-def setup_registration(hass: HomeAssistantType, store: Store,
-                       registration: Dict) -> None:
+async def setup_registration(hass: HomeAssistantType, store: Store,
+                             registration: Dict) -> None:
     """Register the webhook for a registration and loads the app component."""
     registration_name = 'Mobile App: {}'.format(registration[ATTR_DEVICE_NAME])
     webhook_id = registration[CONF_WEBHOOK_ID]
     webhook_register(hass, DOMAIN, registration_name, webhook_id,
                      partial(handle_webhook, store))
 
+    device_registry = await dr.async_get_registry(hass)
+    device_registry.async_get_or_create(
+        config_entry_id=webhook_id,
```

Comment:
This is incorrect. We don't currently have a config_entry_id.

Reply:
What should I be providing instead? Not providing `config_entry_id` causes this:

```
Traceback (most recent call last):
  File "/Users/robbiet480/Repos/HomeAssistant/home-assistant/venv/lib/python3.7/site-packages/aiohttp/web_protocol.py", line 418, in start
    resp = await task
  File "/Users/robbiet480/Repos/HomeAssistant/home-assistant/venv/lib/python3.7/site-packages/aiohttp/web_app.py", line 458, in _handle
    resp = await handler(request)
  File "/Users/robbiet480/Repos/HomeAssistant/home-assistant/venv/lib/python3.7/site-packages/aiohttp/web_middlewares.py", line 119, in impl
    return await handler(request)
  File "/Users/robbiet480/Repos/HomeAssistant/home-assistant/homeassistant/components/http/real_ip.py", line 33, in real_ip_middleware
    return await handler(request)
  File "/Users/robbiet480/Repos/HomeAssistant/home-assistant/homeassistant/components/http/ban.py", line 68, in ban_middleware
    return await handler(request)
  File "/Users/robbiet480/Repos/HomeAssistant/home-assistant/homeassistant/components/http/auth.py", line 206, in auth_middleware
    return await handler(request)
  File "/Users/robbiet480/Repos/HomeAssistant/home-assistant/homeassistant/components/http/view.py", line 112, in handle
    result = await result
  File "/Users/robbiet480/Repos/HomeAssistant/home-assistant/homeassistant/components/http/data_validator.py", line 46, in wrapper
    result = await method(view, request, *args, **kwargs)
  File "/Users/robbiet480/Repos/HomeAssistant/home-assistant/homeassistant/components/mobile_app/http_api.py", line 70, in post
    await setup_registration(hass, self._store, data)
  File "/Users/robbiet480/Repos/HomeAssistant/home-assistant/homeassistant/components/mobile_app/webhook.py", line 68, in setup_registration
    sw_version=registration[ATTR_OS_VERSION]
TypeError: async_get_or_create() missing 1 required keyword-only argument: 'config_entry_id'
```

Reply:
Ah, hmmm . 

Okay, so I think that this means that we should start saving mobile app registrations as config entries.

Reply:
Which would be a new PR. Let's put this PR on hold.

Reply:
That's what I figured!

Reply:
@balloob Okay, so what specifically do I need to be doing for config entries? Am I basically just duplicating what the iOS component is doing right now with `register_discovery_flow`